pages/content/amp-dev/documentation/guides-and-tutorials/start/bento_guide.md

```diff
@@ -0,0 +1,286 @@
+*   [amp-inline-gallery](../../../documentation/components/reference/amp-inline-gallery-v1.0.md)
+    *   amp-inline-gallery-pagination
+    *   amp-inline-gallery-thumbnails
+*   [amp-stream-gallery](../../../documentation/components/reference/amp-stream-gallery-v1.0.md)
+*   [amp-date-countdown](../../../documentation/components/reference/amp-date-countdown-v1.0.md)
+*   [amp-date-display](../../../documentation/components/reference/amp-date-display-v1.0.md)
+*   [amp-fit-text](../../../documentation/components/reference/amp-fit-text-v1.0.md)
+*   [amp-instagram](../../../documentation/components/reference/amp-instagram-v1.0.md)
+*   [amp-lightbox](../../../documentation/components/reference/amp-lightbox-v1.0.md)
+*   [amp-selector](../../../documentation/components/reference/amp-selector-v1.0.md)
+*   [amp-social-share](../../../documentation/components/reference/amp-social-share-v1.0.md)
+*   [amp-timeago](../../../documentation/components/reference/amp-timeago-v1.0.md)
+*   Video components
+    *   [amp-video](../../../documentation/components/reference/amp-video-v1.0.md)
+    *   [amp-youtube](../../../documentation/components/reference/amp-youtube-v1.0.md)
+
+# Bento in action
+
+The example below demonstrates how to include `amp-base-carousel` and `amp-accordion` in a non-AMP HTML page.
+
+```
+<!DOCTYPE html>
+<html ⚡>
+<head>
+  <meta charset="utf-8" />
+  <title>Bento Carousel Example</title>
+  <script async src="https://cdn.ampproject.org/v0.js"></script>
+  <script async custom-element="amp-base-carousel" src="https://cdn.ampproject.org/v0/amp-base-carousel-1.0.js"></script>
+  <script>
+    (self.AMP = self.AMP || []).push(function(AMP) {
+      AMP.toggleExperiment("bento", true);
+    });
+  </script>
+  <style>
+      amp-base-carousel {
+        display: block;
+        height: 400px;
+        width: 600px;
+        overflow: hidden;
+      }
+    </style>
+</head>
+<body>
+  <h1>Bento AMP Components</h1>
+  <p>
+    This demo includes amp-base-carousel with looping and autoadvancing on 5s
+    intervals. The two buttons demonstrate how to apply the events provided
+    through the amp-base-carousel API to elements external to the Bento
+    component.
+  </p>
+  <amp-base-carousel id="my-carousel" controls="never" loop auto-advance auto-advance-interval="5000">
+    <img width="600px" height="400px" src="https://amp.dev/static/samples/img/green_apple_1_1024x682.jpg" />
+    <img width="600px" height="400px" src="https://amp.dev/static/samples/img/golden_apple1_1024x682.jpg" />
+    <img width="600px" height="400px" src="https://amp.dev/static/samples/img/product2_1024x682.jpg" />
+  </amp-base-carousel>
+  <button id="previous-button">
+    Previous Slide
+  </button>
+  <button id="next-button">
+    Next Slide
+  </button>
+  <script>
+    async function nextSlide() {
+      const carousel = await document.querySelector('amp-base-carousel')
```

Comment:
No `await` here. `document.querySelector` is a synchronous function. We could, though, add `await customElements.whenDefined('amp-base-carousel')` if you'd like.

Reply:
I've removed both `awaits`. Is use of `await customElements.whenDefined('amp-base-carousel')` recommended?

Reply:
I think for an example we should give the `await customElements.whenDefined('...')` to make it always functional. Elsewhere we can explain that it's only needed to be run once, but I think it's mostly self-explanatory.

Reply:
Could I get a more complete code example? I tried updating the example to include `await customElements.whenDefined('...')` last week and it continued to throw me the `Cannot read property 'getApi' of undefined` error.

Reply:
Hmm. `await customElements.whenDefined('...')` should be always sufficient against `getApi undefined` errors.

Reply:
I'm able to get it to work with this:
```
<button id="next-button">
Next Slide
</button>
<script>
let nextButton = document.getElementById('next-button')
    function nextSlide(){
      const carousel = document.querySelector('amp-base-carousel');
      customElements
        .whenDefined('amp-base-carousel')
        .then(() => {
          return carousel.getApi();
        })
        .then((api) => {
          // Make the carousel go to the next slide
          api.next();
      });
    }
 nextButton.addEventListener("click", nextSlide())
```

But this throws an error:

```
  <button id="next-button">
    Next Slide
  </button>
  <script>
    async function nextSlide() {
      const carousel = await customElements.whenDefined('amp-base-carousel');
      const api = await carousel.getApi();
      api.next();
    }
    const nextButton = document.querySelector('#next-button');
    nextButton.addEventListener('click', nextSlide);
  </script>
```

Reply:
The first one is correct (or one of possible correct) forms. The second has an error. `customElements.whenDefined` does not return an element, so a further call to `await carousel.getApi()` will fail. If we were to build on the first one with `await`, it'd look like this:

```
async function nextSlide(){
  const carousel = document.querySelector('amp-base-carousel');
  await customElements.whenDefined('amp-base-carousel');
  const api = await carousel.getApi();
  // Make the carousel go to the next slide
  api.next();
}
const nextButton = document.getElementById('next-button')
nextButton.addEventListener("click", nextSlide)
```

(a small nit: notice also that `addEventListener` should just have `nextSlide`, not `nextSlide()`).